.github/workflows/build-local.yml

```diff
@@ -0,0 +1,76 @@
+# Build directly on the GitHub runner
+name: build-local
+on:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: build-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}-${{github.workflow}}
```

Comment:
For the group, I would replace `build-` with `build-local-` or `prebuild-`, to differentiate it from the full build.